dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/harvest/server/OAIRecordServiceBeanTest.java

```diff
@@ -104,6 +105,28 @@ public void updateOaiRecordForDataset_WithoutUpdates() {
 
     }
 
+    @Test
+    public void updateOaiRecordForDataset_WithoutUpdates_EmbargoExpired() {
+        //given
+        oaiRecordServiceBean.setSystemClock(Clock.systemDefaultZone());
+
+        Dataset dataset = setupDatasetData();
+        dataset.setLastChangeForExporterTime(Date.from(Instant.now(Clock.systemDefaultZone()).minus(1, ChronoUnit.DAYS)));
+        dataset.setEmbargoDate(Date.from(dataset.getLastChangeForExporterTime().get().toInstant().plus(12, ChronoUnit.HOURS)));
+
+        HashMap<String, OAIRecord> oaiRecords = new HashMap<>();
+        OAIRecord oaiRecord = setupOaiRecord(oaiRecords);
+        oaiRecord.setRemoved(false);
+
+        //when
+        oaiRecordServiceBean.updateOaiRecordForDataset(dataset,"setName", oaiRecords, Logger.getGlobal());
+
+        //then
+        Assert.assertEquals(Instant.now(Clock.systemDefaultZone()), oaiRecord.getLastUpdateTime().toInstant());
```

Comment:
Passing of this assert is time deterministic (time pass by when executing test). In some cases it can false positively fail and in some cases it can pass with bad `oaiRecordServiceBean.updateOaiRecordForDataset` implementation. That's why other tests use fixed clock.